CRM/Profile/Page/Dynamic.php

```diff
@@ -416,4 +431,30 @@ public function overrideExtraTemplateFileName() {
     return $fileName ? $fileName : parent::overrideExtraTemplateFileName();
   }
 
+  /**
+   * Get the email field as a task link, if not on hold or set to do_not_email.
+   *
+   * @param string $email
+   *
+   * @return string
+   * @throws \API_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  protected function getLinkedEmail($email): string {
+    if (!$email) {
+      return '';
+    }
+    $emailID = Email::get()->setWhere([['contact_id', '=', $this->_id], ['is_primary', '=', TRUE], ['on_hold', '=', FALSE], ['contact.is_deceased', '=', FALSE], ['contact.is_deleted', '=', FALSE], ['contact.do_not_email', '=', FALSE]])->execute()->first()['id'];
```

Comment:
I would suggest to put each condition in its own line like this:

```php
$emailID = Email::get()->setWhere([
  ['contact_id', '=', $this->_id],
  ['is_primary', '=', TRUE],
  ['on_hold', '=', FALSE],
  ['contact.is_deceased', '=', FALSE],
  ['contact.is_deleted', '=', FALSE],
  ['contact.do_not_email', '=', FALSE],
])->execute()->first()['id'];
```

Reply:
But would remove the conditions about primary email and would search for the specific email address.